tests/unittest_brain_dataclasses.py

```diff
@@ -683,3 +687,25 @@ class A:
     inferred = code.inferred()
     assert len(inferred) == 1
     assert isinstance(inferred[0], nodes.ClassDef)
+    assert inferred[0].is_dataclass
+
+
+def test_non_dataclass_is_not_dataclass() -> None:
+    """Test that something that isn't a dataclass has the correct attribute."""
+    ast_nodes = astroid.extract_node(
```

Comment:
```suggestion
    ast_nodes: List[nodes.NodeNG] = astroid.extract_node(  # type: ignore[assignment]
```
`extract_node` can return either a list or an individual node. (Not the best API design 😕) Typing it as `List[nodes.NodeNG]` should help.

Reply:
That's why I added `_extract_single_node` in https://github.com/PyCQA/astroid/pull/1358/files. However, now that I think about it it should probably use an `assert` instead of returning the first element..
We could create `extract_nodes` as well?

Reply:
Something for another day. We have took many open problems already 😅